### PR TITLE
[BugFix] fix file stream time scanning metrics bug

### LINKED ISSUE #5019: fix file stream time scanning metrics bug
**Describe the bug**
currently, file stream time scanning will be 'start()' only once, and may be 'stop()' many times. After the first calling to 'stop()', the self.start value of time scanning will be replaced to default value by 'take()'. The subsequent calling to 'stop()' make no sense.
**To Reproduce**
explain analyze select * from table;
**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by xiaoyong-z**
# Which issue does this PR close?

Closes #5019

# Rationale for this change

currently, file stream time scanning will be 'start()' only once, and may be 'stop()' many times. After the first calling to 'stop()', the self.start value of time scanning will be replaced to default value by 'take()'. The subsequent calling to 'stop()' make no sense.

In this pr, we will 'start()' time scanning again, if the scan of current batch is success.

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/physical_plan/file_format/file_stream.rs [hunk 1]
```diff
@@ -268,7 +268,7 @@ impl<F: FileOpener> FileStream<F> {
                         if result.is_err() {
                             self.state = FileStreamState::Error
                         }
-
+                        self.file_stream_metrics.time_scanning.start();
```
Comment (alamb) [inline datafusion/core/src/physical_plan/file_format/file_stream.rs:271]: The comment for `time_scanning` say

>     /// Time elapsed for file scanning + first record batch of decompression + decoding

Prior to this PR the timer is counting just the time spent until the data starts being produced, but doesn't account for time while the data is being produced.

If we want to change `time_scanning` to include the time while data is being produced, I think we should update the comment as well.

What would you think about making 2 metrics:

```
    /// Time elapsed for file scanning + first record batch of decompression + decoding
    pub time_scanning_until_data: StartableTime,

    /// Total elapsed time for for scanning + record batch decompression / decoding
    pub time_scanning_total: StartableTime,
```

**Review by alamb — COMMENTED** [review]
Thank you for all these contributions @xiaoyong-z  -- very nice 👍

**Comment by alamb** [comment]
cc @tustvold

**Comment by xiaoyong-z** [comment]
cc @tustvold @alamb

**Review by alamb — APPROVED** [review]
LGTM -- thank you @xiaoyong-z

**Pull request merged**